drivers/adc/max9611pmb/max9611.h

```diff
@@ -0,0 +1,197 @@
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#ifndef __MAX9611_H__
+#define __MAX9611_H__
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdint.h>
+#include <stdlib.h>
+#include "no_os_i2c.h"
+#include "no_os_util.h"
+#include "no_os_error.h"
+#include "no_os_delay.h"
+
+/******************************************************************************/
+/************************** max538x Definitions *******************************/
+/******************************************************************************/
+#define MAX9611_MUX_MASK		NO_OS_GENMASK(2, 0)
+#define MAX9611_MODE_MASK		NO_OS_GENMASK(7, 5)
+
+#define MAX9611_SHDN_MASK		NO_OS_BIT(3)
+#define MAX9611_LR_MASK			NO_OS_BIT(4)
+#define MAX9611_DTIME_MASK		NO_OS_BIT(3)
+#define MAX9611_RTIME_MASK		NO_OS_BIT(2)
+
+#define MAX9611_OUT_OF_BOUNDS  	0x0C
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+enum max9611_regs {
```

Comment:
usually, the registers are declared using `define` macros, which improves the readability a bit.

Reply:
Removed in latest commit.

Reply:
Ok, I won't push too hard on this, but how is this different from using macros for the register addresses and keeping the out of bounds check? You can still pass integers instead of enums...

Reply:
Rewrote addresses as macros and just implemented bounds checking in the read and write functions.